django_datawatch/querysets.py

```diff
@@ -9,8 +9,10 @@
 
 class ResultQuerySet(models.QuerySet):
     def for_user(self, user):
-        return self.filter(Q(assigned_to_group__isnull=True) | Q(assigned_to_group__in=user.groups.all()),
-                           Q(assigned_to_user__isnull=True) | Q(assigned_to_user=user))
+        user_groups = user.groups.all()
+        return self.filter(
+            Q(assigned_groups__in=user_groups) | Q(assigned_users=user),
```

Comment:
Isn't this now excluding the results that have no user and no group assigned?
Would you mind adding tests for that case?